linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -553,6 +569,23 @@ private[emitter] final class KnowledgeGuardian(config: Emitter.Config) {
           _.methods.filter(_.value.flags.namespace == MemberNamespace.Public))
     }
 
+    private def computeHijackedDescendants(
+        hijackedClasses: Iterable[LinkedClass]): Map[ClassName, List[ClassName]] = {
+      val pairs = for {
+        hijackedClass <- hijackedClasses
+        ancestor <- hijackedClass.ancestors
+        if ancestor != hijackedClass.className
+      } yield {
+        (ancestor, hijackedClass)
+      }
+
+      for {
+        (ancestor, pairs) <- pairs.groupBy(_._1)
+      } yield {
+        (ancestor, pairs.map(_._2.className).toList)
```

Comment:
We should make sure that the `List`s on the right are always ordered in the same way. Otherwise, we could get spurious invalidate-all events just because the returned lists are not equal (although they contain the same elements).